holoviews/core/util.py

```diff
@@ -24,6 +26,57 @@
 except ImportError:
     dd = None
 
+
+
+
+class HashGenerator(json.JSONEncoder):
+    """
+    Extends JSONEncoder to generate a hashable string for as many types
+    of key as possible. These keys are supplied by widgets and Stream
+    parameters and may be nested. If an unrecognized object type is
```

Comment:
Hmm, all strings are hashable already. It is about taking an arbitrary thing and generating a string from it. I agree that is a less ugly name though... @philippjfr Any suggestions?